### PR TITLE
fix(tailwind): fix webpack cache not work when using twin.macro

### PR DESCRIPTION
**Pull request by chenjiahan**
## Description

Fix webpack cache not work when using twin.macro, the temporary twin config file will invalidate webpack cache:

## Types of changes

- [ ] Docs change / Dependency upgrade
- [x] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/toolkit/utils/src/compatRequire.ts [hunk 1]
```diff
@@ -44,3 +44,12 @@ export const cleanRequireCache = (filelist: string[]) => {
     delete require.cache[filepath];
   });
 };
+
+export function deleteRequireCache(path: string) {
+  if (require.cache[path]) {
+    delete require.cache[path];
+  }
+  if (module.children) {
```
Comment (targeral) [inline packages/toolkit/utils/src/compatRequire.ts:52]: 这里如果不这样处理会有什么问题呢？

File: packages/cli/plugin-tailwind/src/cli.ts [hunk 2]
```diff
@@ -116,13 +118,32 @@ export default (
...
+                // https://github.com/ben-rogerson/twin.macro/releases/tag/3.0.0
+                const twinMajorVersion = getTwinMacroMajorVersion(appDirectory);
+                const supportConfigObject =
+                  twinMajorVersion && twinMajorVersion >= 3;
+
+                let twinConfig: string | Record<string, unknown> =
+                  internalTwConfigPath;
+
+                if (supportConfigObject) {
+                  twinConfig = require(internalTwConfigPath);
```
Comment (targeral) [inline packages/cli/plugin-tailwind/src/cli.ts:132]: 这里复用 98 行（postcss 函数里面使用过）的 `getTailwindConfig` 函数怎么样？

File: packages/toolkit/utils/src/compatRequire.ts (hunk 1 above)
Comment (chenjiahan) [reply packages/toolkit/utils/src/compatRequire.ts:52]: 如果不处理 children 的话，webpack 还是会把这个模块纳入 cache 检测范围，所以 children 也要处理下

File: packages/cli/plugin-tailwind/src/cli.ts (hunk 2 above)
Comment (chenjiahan) [reply packages/cli/plugin-tailwind/src/cli.ts]: 确实用 `getTailwindConfig` 更好些，已修改实现👌🏻

File: packages/toolkit/utils/src/compatRequire.ts (hunk 1 above)
Comment (targeral) [reply packages/toolkit/utils/src/compatRequire.ts:52]: 学习到了

**Review by targeral — APPROVED** [review]

**Pull request merged**